types/react/index.d.ts

```diff
@@ -308,11 +374,6 @@ declare namespace React {
         // on the existence of `children` in `P`, then we should remove this.
         readonly props: Readonly<{ children?: ReactNode }> & Readonly<P>;
         state: Readonly<S>;
-        /**
-         * @deprecated
-         * https://reactjs.org/docs/legacy-context.html
-         */
-        context: any;
```

Comment:
Hm, it is strange, cause here https://reactjs.org/docs/context.html#classcontexttype we can see, React allows us to use just this.context, but it is not possible, if context will be deleted from types of Component.

Reply:
By the way, there is no anything like value in Component.contextType.Provider.

Reply:
The point is to force the user to declare it themselves because there is nothing better we can do other than `any` here, and having it already declared would allow people to access it even without a `static contextType`.

`value` is the `value` prop of the `Provider` exotic component.

Reply:
Hm, could you explain, why this code is not working:
```
class Component<P, S> {
    static contextType: Context<any>;
    ...
    ...
    context: typeof Component.contextType extends Context<infer T> ? T : never
}
```

Type of context is any always( But it works perfectly in my own component.

Reply:
That is because, by declaring it in the superclass, you don't have access to its type as assigned in the subclass. The type of the constructor itself is not generic so it just uses the type as declared, assuming `Component` is constructed as is and not through a subclass. (Note: despite how I phrased it, this still wouldn't work if `Component` was `abstract`)

This is the deficiency in TypeScript's class declaration language: it is possible to make an instance type generic, but not the constructor object itself generic.

There is a possible hack with constructible interfaces, though, if they're even accepted in an `extends` clause. It could be possible to just completely replace the `class Component` declaration with a pair of `interfaces` and a `const`. But... at what cost?.

Reply:
Thank you for explanation. I agree, what it is ok to get rid of context from typings.